mesonbuild/build.py

```diff
@@ -810,7 +810,7 @@ def process_compilers(self):
                         break
                 else:
                     if is_known_suffix(s):
-                        raise MesonException('No {} machine compiler for "{}"'.
+                        raise MesonException('No {} machine compiler for "{}". Check the documentation of `project` for how to specify required compilers!'.
```

Comment:
This definitely needs improving, but I'm not sure this is how:

* The documentation for `project` doesn't mention compilers.  It mentions languages (which have compilers)
* Languages can also be added with `add_language`
* You can probably also get this error in cross-situations where the language has been specified, but the compiler isn't required (this is the 'The build machine compilers for the specified languages are not required' sentence in the documentation for `project`) and isn't present, but you still try to use it. (see https://github.com/mesonbuild/meson/pull/6532#issuecomment-583859162 for the gruesome details)